### ISSUE TITLE
Fido2 Conformance Test Tool : authenticatorMakeCredential doesn't be cancelled or wait for response in HID Transport Test P-9 

### ISSUE THREAD

**Issue by joshuajylin** [fido-alliance/conformance-tools-issues#338]
By submitting this issue you are acknowledging that any information regarding this issue will be publicly available.

If you have privacy concerns, please email [conformance-tools@fidoalliance.org](mailto:conformance-tools@fidoalliance.org)

## What protocol and version of the protocol are you testing?

FIDO2 v1.0

## What is your implementation class?

ASM+Authr

## What is the version of the tool are you using?
v0.10.105

## What is the OS and the version are you running?
macOS 10.13.5

## Issue description
HID Transports test P-9 doesn't finish authenticatorMakeCredential.
I don't know what is the correct steps in HID Transports test P-9, but it should cancel authenticatorMakeCredential or wait for response, instead of starting P-10 immediately.
Current behavior will cause P-10 got incorrect response.

**Comment by herrjemand** [comment]
@joshuajylin At the end of the test P-9 send cancel request to the authenticator

**Comment by joshuajylin** [comment]
@herrjemand yeah, I saw  "window.navigator.fido.fido2conformancetest.hid.sendHIDCancel(getDeviceInfo(), NEWCID)".
But I doesn't see a cancel request received from the host. Is it CTAPHID_CANCEL?
Could you take a look for ensuring the cancel request will be sent successfully?

**Comment by herrjemand** [comment]
HID DATA SENT: ffffffff8600080103538c60797b4400000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
HID DATA RECEIVED: ffffffff8600110103538c60797b4400000002020100010300000000000000000000000000000000000000000000000000000000000000000000000000000000

HID DATA SENT: 0000000290012a01a40158200baf763b6901d0cbcb946d6858980a0f7c1ca56295ba96c67a74d6c79b11db9c02a26269647468747470733a2f2f616c6f6f6670
0000000200696e652e7675646e616d657829546865204578616d706c6520436f72706f726174696f6e20776974682066616b6520646f6d61696e2103a4626964
00000002015820af6c4c5ebad8ecc54da7e2cc5a83e35173bd7b7ca2b9fc310d1896ec39d5e6196469636f6e782868747470733a2f2f706963732e61636d652e
0000000202636f6d2f30302f702f61426a6a6a707150622e706e67646e616d6577677265746368656e6d6f40616c6f6f6670696e652e76756b646973706c6179
00000002034e616d656b477265746368656e204d6f0482a263616c672664747970656a7075626c69632d6b6579a263616c6739010064747970656a7075626c69
0000000204632d6b6579000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
HID DATA READ: 00000002bb0001020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000

HID DATA SENT: 00000002910000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
HID DATA READ: 00000002bb0001020000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000

**Comment by herrjemand** [comment]
<img width="1032" alt="screenshot 2018-07-26 12 56 33" src="https://user-images.githubusercontent.com/1636116/43261079-57721460-90d3-11e8-85d3-e16e1e78191f.png">

**Comment by joshuajylin** [comment]
@herrjemand 
![no_cancel copy](https://user-images.githubusercontent.com/1489319/43263280-6517e080-9115-11e8-8a27-8d14f952839d.png)

**Comment by herrjemand** [comment]
@joshuajylin can  you try v0.10.915?

**Comment by joshuajylin** [comment]
@herrjemand I have tried v0.10.915, host can send out CTAPHID_CANCEL.
But there is another issue about P-10.
"return timeoutPromise(100)" will cause that the response of CTAPHID_CANCEL fails to be received, no matter the timeout value is greater or less than 100.
After I removed "timeoutPromise(100)", P-10 passed.

<img width="1272" alt="screen shot 2018-07-31 at 11 08 33 am" src="https://user-images.githubusercontent.com/1489319/43436384-e4db1d4c-94b6-11e8-997f-ed617e3c08c7.png">
